### ISSUE TITLE
Regarding training of a modified base model

### ISSUE THREAD

**Issue by hukai916** [nanoporetech/megalodon#268]
Hi developers,

Thanks for creating megalodon! Not sure if it is okay to ask a usage related question here. 

I found it hard to follow the documentation regarding the training of a modified base model. By reading the tutorial, I understand that there are two sample types currently supported:
    1. Modified base in known sequence context (e.g. bacterial methylation)
    2. Native (fractionally modified) sample with existing modified base basecalling model

I believe my sample belongs the the second type where I have ground truth data for targeted sites that are supposed to contain modified bases. However, seems that all the example codes are related to the first scenario. Besides that, I have the following questions:
    1. To train the model, what kind of existing modified basecalling model shall I use given that I am trying to model a new modification? 
    2. Shall I follow `Bootstrap Modified Base Annotation` or the `Ground Truth Aided Bootstrap Modified Base Annotation` section? 
    3. For the latter, what does `ground_truth_methylation.CG.bed` suppose to contain and how to generate one? 
    4. Where can I find example data?

I apologize for these many questions, I have been scratching my heads for quite a few days, hope you can help.

Thanks,

--Kai

**Comment by marcus1487** [comment]
First of all I would recommend switching over to Remora modified baes models. This is the new state of the art model style and greatly simplifies the training data prep and training processes.

Within the megalodon setting it does sound like you have a sample outside of the two example settings. You'll have to expand a bit more on your training data in order to provide useful advice on if this data is likely to be sufficient and how training would proceed.

I'll try to address your specific questions which may inform your training procedure.

1. The existing model must be of the same modification type as the final model. This initial model may be of lower quality, but have some statistical power to detect the modified base, or the initial model may be stronger in particular sequence contexts, but again show statistical power to detect the modification in other contexts.
2. This requires more information about the training data sample type.
3. The ground_truth_methylation.bed file contains the ground truth fraction of modified bases at each reference position. This is generally obtained from an alternative data source (e.g. bisulfite for 5mC), though note that this is no longer required in Remora style models as bisulfite data is quite error prone (and we have recently presented evidence that Remora models are more accurate than bisulfite). 
4. We do not have official example training data released at this time, but there are public datasets which may suffice for training purposes.

**Comment by hukai916** [comment]
Hi @marcus1487,

Thanks very much for the informative answer! I was exploring Remora recently per your suggestion. 

Let me expand a bit on our data setting so that you can confirm with me if our data fits into either of the 2 sample types. Basically, we have a specific modification, T -> U, that we would like to train a custom model to detect. To get the ground truth, we barcoded (10bp) the 4 canonical bases and the modified base and prepared the library by synthesis. The library construction looks like below:
```
bc1-bc2-bc3-bc4-bc5-backbone-base5-base4-base3-base2-base1
```
We then Nanopore sequenced the construction, and by de-barcoding, we can determine the identities of the bases incorporated. Our preliminary analysis confirms that the raw signal from modified U differ dramatically from T or other canonical bases. Therefore, we think it could be a good dataset for training of a custom model with Megalodon or Remora. Can you let me know if our data fit into any of the data types (we don't have a specific sequence context for the modified base)?

Thanks a lot!

--Kai

**Comment by marcus1487** [comment]
This sounds a bit closer to a research project we are working on internally to use synthetic samples to train remora models. See [this thread](https://github.com/nanoporetech/remora/issues/15) for more details and limitations of this idea and related work. I'm not sure I completely follow the setup you have described here. I'm not sure exactly what purpose the barcodes server here. If you could add some more detail I'd be happy to assist further.

**Comment by hukai916** [comment]
I would like to provide more info, however, this is a collaborative project and I may need to get some approval before sharing more info. What I can say here is that the barcode is to determine the identities of inserted bases on the backbone.

Pls let me know if you see a fit for our data to use Megalodon, and I likely can provide more info and will keep you posted.
